plugin/evm/header/block_gas_cost.go

```diff
@@ -11,21 +11,27 @@ import (
 	"github.com/ava-labs/coreth/params/extras"
 	"github.com/ava-labs/coreth/plugin/evm/upgrade/ap4"
 	"github.com/ava-labs/coreth/plugin/evm/upgrade/ap5"
+	"github.com/ava-labs/libevm/common"
 )
 
 var (
 	errBaseFeeNil        = errors.New("base fee is nil")
 	errBlockGasCostNil   = errors.New("block gas cost is nil")
 	errExtDataGasUsedNil = errors.New("extDataGasUsed is nil")
+	errNoGasUsed         = errors.New("no gas used")
 )
 
 // BlockGasCost calculates the required block gas cost based on the parent
 // header and the timestamp of the new block.
+// Prior to AP4, the returned block gas cost will be nil.
 func BlockGasCost(
 	config *extras.ChainConfig,
 	parent *types.Header,
```

Comment:
what's our typical process of using header vs headerExtra? This is pretty much our own code, I think we can require headerExtra here. but certainly not a big issue.